cdc/model/mounter.go

```diff
@@ -108,16 +108,48 @@ const (
 
 // ResolvedTs is the resolved timestamp of sink module.
 type ResolvedTs struct {
-	Ts   uint64
-	Mode ResolvedMode
+	Mode    ResolvedMode
+	Ts      uint64
+	BatchID uint64
 }
 
-// NewResolvedTs creates a new ResolvedTs.
+// NewResolvedTs creates a normal ResolvedTs.
 func NewResolvedTs(t uint64) ResolvedTs {
-	return ResolvedTs{Ts: t, Mode: NormalResolvedMode}
+	return ResolvedTs{Ts: t, Mode: NormalResolvedMode, BatchID: math.MaxUint64}
+}
+
+// IsBatchMode returns true if the resolved ts is BatchResolvedMode.
+func (r ResolvedTs) IsBatchMode() bool {
+	return r.Mode == BatchResolvedMode
+}
+
+// ParseTs returns a timestamp `ts` based on the r.mode, which means that all events
+// whose commitTs is less than or equal to `ts` are sent to Sink.
+func (r ResolvedTs) ParseTs() uint64 {
+	switch r.Mode {
+	case NormalResolvedMode:
+		// with NormalResolvedMode, cdc guarantees all events whose commitTs is
+		// less than or equal to `resolved.Ts` are sent to Sink.
+		return r.Ts
+	case BatchResolvedMode:
+		// with BatchResolvedMode, cdc guarantees all events whose commitTs is
+		// less than `resolved.Ts` are sent to Sink.
+		return r.Ts - 1
+	default:
+		log.Error("unknown resolved mode", zap.Any("resolved", r))
+		return 0
+	}
+}
+
+// EqualOrGreater judge whether the resolved ts is equal or greater than the given ts.
+func (r ResolvedTs) EqualOrGreater(r1 ResolvedTs) bool {
+	if r.Ts == r1.Ts {
+		return r.BatchID >= r1.BatchID
+	}
+	return r.Ts > r1.Ts
```

Comment:
`EqualOrGreater` -> `>=`. 

L149, shall we use `>=` ?

Reply:
We have already considered the equal case in line 146, so it is unnecessary to judge again. Besides, does using `>=` make the semantics more explicit?